docs/ARDUPILOT.md

```diff
@@ -23,19 +23,61 @@ For my Matek H743 board the configuration is:
+
+## RSSI
+
+RSSI is a relatively confusing topic with ArduPilot, since there are multiple values and sources.
+
+First, a comment on how RSSI is represented by ArduPilot. ArduPilot converts any RSSI value into the range [0.0...1.0], where 0.0 represents the weakest and 1.0 the strongest signal. The equation which is used for this scaling depends on the source of the RSSI value. In the dataflash log this value is available as 'RSSI:RXRSSI'. For other uses, this internal RSSI value is then further converted into whatever the output format demands. For instance, the RSSI value that is displayed in the OSD uses the internal value multiplied by 100 which is displayed as 0% ... 100%. For MAVLink it is converted into the range [0...254] (0 = weakest signal, 254 = strongest signal, 255 = undefined)(from here on we'll refer to this scaling as MAVLink units for RSSI).
+
+The source of the RSSI value (and thus applied conversion formula) is determined by the ArduPilot parameter 'RSSI_TYPE'. With mLRS these settings can be relevant:
+- RSSI_TYPE = 3 (ReceiverProtocol): mLRS will send RSSI in dBm in the CRSF stream and ArduPilot will apply scaling using the following formula: RSSI = 1 - ((RSSI_dBM - 50) / 70). This means that -120 dBm is mapped to 0.0 and -50 dBm to 1.0.
+    - ***Note***: When using RSSI_TYPE = 3, it is possible to replace the RSSI with LQ using the RC_OPTIONS bitmask.
+- RSSI_TYPE = 5 (TelemetryRadioRSSI): mLRS will send RSSI in the MAVLink stream to the ArduPilot via the RADIO_STATUS message, in MAVLink units. mLRS applies the formula RSSI = (1 - ((RSSI_dBM - 50) / 70))*254 to convert to MAVLink units.
+- RSSI_TYPE = 2 (RCChannelPwmValue): mLRS can be configured to send RSSI (and LQ) as a channel value to ArduPilot. This option is not recommended and not further described.
+
+MissionPlanner gets receiver RSSI data from ArduPilot via MAVLink messages (usually via the 'RC_CHANNELS' message) in the range [0...254] (or 255 for undefined). This value is available in MissionPlanner as 'rxrssi'.
+
+In addition a Tx module may send 'RADIO_STATUS' MAVLink messages to MissionPlanner, with RSSI in MAVLink units. The data from the Tx module is available in MissionPlanner as 'rssi' and 'remrssi' (also 'noise' and 'remnoise' are available).
+
+This is enabled in mLRS when the parameter 'Tx Snd RadioStat' is set to '1 Hz'. mLRS will use the following formula to convert from RSSI in dBm to MAVLink units: RSSI = (1 - ((RSSI_dBM - 50) / 70)) * 254.
+
+Therefore in Mission Planner three RSSI metrics are available when using mLRS (all in MAVLink units):
+
+1. rxrssi - the RSSI of the mLRS receiver, as seen by ArduPilot (and passed back to the ground via 'RC_CHANNELS' messages)
+2. rssi - the RSSI of the mLRS Tx module
+3. remrssi - the RSSI of the mLRS receiver, as seen by the mLRS Tx module (passed back to the ground as part of mLRS' OTA data packets)
```

Comment:
Prefer 'passed back to the ground as part of the mLRS OTA)

Reply:
it's not a sentence then, OTA stands for over the air IMHO
so it is as part of the mLRS over the air
?

Reply:
How about '...(passed back to the ground within the mLRS packet)'